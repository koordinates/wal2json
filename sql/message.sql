\set VERBOSITY terse

-- predictability
SET synchronous_commit = on;

SELECT 'init' FROM pg_create_logical_replication_slot('regression_slot', 'wal2json');

SELECT 'msg1' FROM pg_logical_emit_message(true, 'wal2json', 'this is a\ message');
SELECT 'msg2' FROM pg_logical_emit_message(false, 'wal2json', 'this is "another" message');

SELECT 'msg3' FROM pg_logical_emit_message(false, 'wal2json', E'\\x31320033003435'::bytea);
SELECT 'msg4' FROM pg_logical_emit_message(false, 'wal2json', E'\\xC0FFEE00C0FFEE'::bytea);
SELECT 'msg5' FROM pg_logical_emit_message(false, 'wal2json', E'\\x01020300101112'::bytea);

BEGIN;
SELECT 'msg6' FROM pg_logical_emit_message(true, 'wal2json', 'this message will not be printed');
SELECT 'msg7' FROM pg_logical_emit_message(false, 'wal2json', 'this message will be printed even if the transaction is rollbacked');
ROLLBACK;

BEGIN;
SELECT 'msg8' FROM pg_logical_emit_message(true, 'wal2json', 'this is message #1');
SELECT 'msg9' FROM pg_logical_emit_message(false, 'wal2json', 'this message will be printed before message #1');
SELECT 'msg10' FROM pg_logical_emit_message(true, 'wal2json', 'this is message #2');
COMMIT;

SELECT data FROM pg_logical_slot_get_changes('regression_slot', NULL, NULL, 'pretty-print', '1');

SELECT 'stop' FROM pg_drop_replication_slot('regression_slot');
