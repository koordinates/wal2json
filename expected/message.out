\set VERBOSITY terse
-- predictability
SET synchronous_commit = on;
SELECT 'init' FROM pg_create_logical_replication_slot('regression_slot', 'wal2json');
 ?column? 
----------
 init
(1 row)

SELECT 'msg1' FROM pg_logical_emit_message(true, 'wal2json', 'this is a\ message');
 ?column? 
----------
 msg1
(1 row)

SELECT 'msg2' FROM pg_logical_emit_message(false, 'wal2json', 'this is "another" message');
 ?column? 
----------
 msg2
(1 row)

SELECT 'msg3' FROM pg_logical_emit_message(false, 'wal2json', E'\\x31320033003435'::bytea);
 ?column? 
----------
 msg3
(1 row)

SELECT 'msg4' FROM pg_logical_emit_message(false, 'wal2json', E'\\xC0FFEE00C0FFEE'::bytea);
 ?column? 
----------
 msg4
(1 row)

SELECT 'msg5' FROM pg_logical_emit_message(false, 'wal2json', E'\\x01020300101112'::bytea);
 ?column? 
----------
 msg5
(1 row)

BEGIN;
SELECT 'msg6' FROM pg_logical_emit_message(true, 'wal2json', 'this message will not be printed');
 ?column? 
----------
 msg6
(1 row)

SELECT 'msg7' FROM pg_logical_emit_message(false, 'wal2json', 'this message will be printed even if the transaction is rollbacked');
 ?column? 
----------
 msg7
(1 row)

ROLLBACK;
BEGIN;
SELECT 'msg8' FROM pg_logical_emit_message(true, 'wal2json', 'this is message #1');
 ?column? 
----------
 msg8
(1 row)

SELECT 'msg9' FROM pg_logical_emit_message(false, 'wal2json', 'this message will be printed before message #1');
 ?column? 
----------
 msg9
(1 row)

SELECT 'msg10' FROM pg_logical_emit_message(true, 'wal2json', 'this is message #2');
 ?column? 
----------
 msg10
(1 row)

COMMIT;
SELECT data FROM pg_logical_slot_get_changes('regression_slot', NULL, NULL, 'pretty-print', '1');
                                                  data                                                   
---------------------------------------------------------------------------------------------------------
 {                                                                                                      +
         "change": [                                                                                    +
                 {                                                                                      +
                         "kind": "message",                                                             +
                         "transactional": true,                                                         +
                         "prefix": "wal2json",                                                          +
                         "content": "this is a\\ message"                                               +
                 }                                                                                      +
         ]                                                                                              +
 }
 {                                                                                                      +
         "change": [                                                                                    +
                 {                                                                                      +
                         "kind": "message",                                                             +
                         "transactional": false,                                                        +
                         "prefix": "wal2json",                                                          +
                         "content": "this is \"another\" message"                                       +
                 }                                                                                      +
         ]                                                                                              +
 }
 {                                                                                                      +
         "change": [                                                                                    +
                 {                                                                                      +
                         "kind": "message",                                                             +
                         "transactional": false,                                                        +
                         "prefix": "wal2json",                                                          +
                         "content": "12"                                                                +
                 }                                                                                      +
         ]                                                                                              +
 }
 {                                                                                                      +
         "change": [                                                                                    +
                 {                                                                                      +
                         "kind": "message",                                                             +
                         "transactional": false,                                                        +
                         "prefix": "wal2json",                                                          +
                         "content": ""                                                                  +
                 }                                                                                      +
         ]                                                                                              +
 }
 {                                                                                                      +
         "change": [                                                                                    +
                 {                                                                                      +
                         "kind": "message",                                                             +
                         "transactional": false,                                                        +
                         "prefix": "wal2json",                                                          +
                         "content": "\u0001\u0002\u0003"                                                +
                 }                                                                                      +
         ]                                                                                              +
 }
 {                                                                                                      +
         "change": [                                                                                    +
                 {                                                                                      +
                         "kind": "message",                                                             +
                         "transactional": false,                                                        +
                         "prefix": "wal2json",                                                          +
                         "content": "this message will be printed even if the transaction is rollbacked"+
                 }                                                                                      +
         ]                                                                                              +
 }
 {                                                                                                      +
         "change": [                                                                                    +
                 {                                                                                      +
                         "kind": "message",                                                             +
                         "transactional": false,                                                        +
                         "prefix": "wal2json",                                                          +
                         "content": "this message will be printed before message #1"                    +
                 }                                                                                      +
         ]                                                                                              +
 }
 {                                                                                                      +
         "change": [                                                                                    +
                 {                                                                                      +
                         "kind": "message",                                                             +
                         "transactional": true,                                                         +
                         "prefix": "wal2json",                                                          +
                         "content": "this is message #1"                                                +
                 }                                                                                      +
                 ,{                                                                                     +
                         "kind": "message",                                                             +
                         "transactional": true,                                                         +
                         "prefix": "wal2json",                                                          +
                         "content": "this is message #2"                                                +
                 }                                                                                      +
         ]                                                                                              +
 }
(8 rows)

SELECT 'stop' FROM pg_drop_replication_slot('regression_slot');
 ?column? 
----------
 stop
(1 row)

